File: app/src/main/java/com/example/kaliumapp/data/database/entities/DailyWaterSummary.kt
package com.example.kaliumapp.data.database.entities

import androidx.room.*

@Entity(
    tableName = "daily_water_summary",
    foreignKeys = [ForeignKey(
        entity = User::class,
        parentColumns = ["id"],
        childColumns = ["userId"],
        onDelete = ForeignKey.CASCADE
    )],
    indices = [
        Index(value = ["userId"]),
        Index(value = ["userId", "date"], unique = true) // Prevent duplicate entries for same user and date
    ]
)
data class DailyWaterSummary(
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "summary_id")
    val summaryId: Int = 0,

    @ColumnInfo(name = "userId")
    val userId: Int,

    @ColumnInfo(name = "date")
    val date: String, // Format: "yyyy-MM-dd"

    @ColumnInfo(name = "totalIntakeMl")
    val totalIntakeMl: Int = 0,

    @ColumnInfo(name = "remainingMl")
    val remainingMl: Int = 0,

    @ColumnInfo(name = "status")
    val status: String = "Active", // Active, Completed, Paused

    @ColumnInfo(name = "created_at")
    val createdAt: Long = System.currentTimeMillis(),

    @ColumnInfo(name = "updated_at")
    val updatedAt: Long = System.currentTimeMillis()
)

File: models/daily_water_summary.py
from sqlalchemy import Column, Integer, String, ForeignKey
from sqlalchemy.orm import relationship
from database.database import Base

class DailyWaterSummary(Base):
    __tablename__ = "daily_water_summary"

    summary_id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    date = Column(String, nullable=False)  # Format: YYYY-MM-DD
    total_ml = Column(Integer, default=0)

    user = relationship("User", back_populates="water_summaries")
    water_intakes = relationship("WaterIntake", back_populates="summary", cascade="all, delete-orphan")